src/bluetooth/TemperatureLogManager.js

```diff
@@ -0,0 +1,83 @@
+import moment from 'moment';
+
+// Default temperature log generator.
+const createTemperatureLog = ({ id, sensor, timestamp, temperature, logInterval }) => ({
+  id,
+  sensor,
+  timestamp: new Date(timestamp),
+  temperature,
+  logInterval,
+});
+
+export class TemperatureLogManager {
+  constructor(dbService, utils) {
+    this.db = dbService;
+    this.utils = utils;
+
+    this.createLogFunc = createTemperatureLog;
+  }
+
+  set createLog(createLogFunc) {
+    this.createLogFunc = createLogFunc;
+  }
+
+  get createLog() {
+    return this.createLogFunc;
+  }
+
+  // Calculates the number of sensor logs that should be saved from some given starting
+  // point. Where the starting point is the timestamp for the next log.
+  calculateNumberOfLogsToSave = (
+    nextPossibleLogTime = 0,
+    logInterval,
+    timeNow = moment().unix()
```

Comment:
Just having this whole class work from unix timestamps (in secs) as I find it pretty easy (Dates suck!) and seems to be a pretty easy format to convert into whatever format is required by the caller (Mobile actually uses `Date` objects`)